tools/clang/lib/SPIRV/LowerTypeVisitor.cpp

```diff
@@ -371,6 +371,21 @@ const SpirvType *LowerTypeVisitor::lowerType(QualType type,
   // AST vector/matrix types are TypedefType of TemplateSpecializationType. We
   // handle them via HLSL type inspection functions.
 
+  // When the memory layout rule is FxcCTBuffer, typeNxM matrix with M > 1 and
+  // N == 1 is a matrix with M rows of the vector with type and size 1. Since
+  // SPIR-V does not have a vector with size 1, we have to use an array.
+  // We have the same rule for column_major typeNxM and row_major typeMxN.
+  if (rule == SpirvLayoutRule::FxcCTBuffer && hlsl::IsHLSLMatType(type)) {
```

Comment:
I'm a bit worried about this code. I need to think about it a bit more. The reason I'm concerned is because we have `isVectorType` function which returns true for such cases, and we have `isMatrixOrArrayOfMatrix` which returns false for these cases. And throughout the codebase we have used these functions assuming that mat1xM or matNx1 are *not* matrices.